src/node/update.ts

```diff
@@ -106,12 +106,13 @@ export class UpdateProvider {
         const httpx = uri.startsWith("https") ? https : http
         const client = httpx.get(uri, { headers: { "User-Agent": "code-server" } }, (response) => {
           if (!response.statusCode || response.statusCode < 200 || response.statusCode >= 400) {
+            response.destroy()
             return reject(new Error(`${uri}: ${response.statusCode || "500"}`))
           }
 
           if (response.statusCode >= 300) {
-            ++redirects
             response.destroy()
+            ++redirects
```

Comment:
What's with these being moved? Feels strangely out of scope

Reply:
@code-asher and I now separate things cleanly into separate commits so it's best to review our PRs in commit order. The reasoning for this is in commit #2 `update.ts: Fix response memory leak` and is for consistency with the `response.destroy` I added above.

ik it's random and can go into another PR but just reducing churn.